ui/pages/connected-accounts/connected-accounts.container.js

```diff
@@ -21,6 +22,24 @@ const mapStateToProps = (state) => {
   const { activeTab } = state;
   const accountToConnect = getAccountToConnectToActiveTab(state);
   const connectedAccounts = getOrderedConnectedAccountsForActiveTab(state);
+  const internalAccounts = getInternalAccounts(state);
+  // Temporary fix until https://github.com/MetaMask/metamask-extension/pull/21553
+  const connectedAccountsWithName = connectedAccounts.map((account) => {
+    const updatedAccount = {
+      ...account,
+      name: internalAccounts.find(
```

Comment:
Simular slight note about optimization here as above
```typescript
const internalAccountsMap = new Map(
    internalAccounts.map((acc) => [acc.address, acc]),
  );
const connectedAccountsWithName = connectedAccounts.map((account) => {
     const updatedAccount = {
       ...account,
       name: internalAccountsMap.get(account.address)?.metadata.name,
     };
     return updatedAccount;
   });

Reply:
Same as the other comment